index.ts

```diff
@@ -386,6 +386,13 @@ class MediaConnection {
         })
 
         this.media.on('message', (message: any) => this.parseMessage(message))
+
+        this.media.on('close', () => {
+          // Sony cast speakers CLOSE when powered down, and don't send updates
+          // Update internal state
+          this.monitor.setPlayState('IDLE');
+          this.monitor.setMedia({});
```

Comment:
You can't call setMedia with {}, it requires an artist and a title. Right now it is only called when a new song starts, not when there is nothing playing.

Also, please remove the semicolon at the end of the line.

Reply:
Will update the semicolons.

Regarding `setMedia`, in my client app, I'm trying to disambiguate between these two states: `IDLE`, `PAUSED`.  In both situations, the shown playState is `pause`.

This is tricky, because in `IDLE` state, the media can't be resumed using `playDevice()`.  I use the empty `media` object to indicate that the player isn't able to resume.

Perhaps this isn't the best approach.  Do you think there's a better play state?

Reply:
Perhaps we should disambiguate `IDLE` to `stop` rather than `pause`?